core.go

```diff
@@ -53,9 +54,49 @@ func (c Core) ListObjectsV2(bucketName, objectPrefix, continuationToken string,
 	return c.listObjectsV2Query(bucketName, objectPrefix, continuationToken, fetchOwner, delimiter, maxkeys)
 }
 
+// CopyObject - copies an object from source object to destination object on server side.
+func (c Core) CopyObject(sourceBucket, sourceObject, destBucket, destObject string, metadata map[string]string) (ObjectInfo, error) {
+	return c.copyObjectDo(context.Background(), sourceBucket, sourceObject, destBucket, destObject, metadata)
+}
+
 // PutObject - Upload object. Uploads using single PUT call.
 func (c Core) PutObject(bucket, object string, data io.Reader, size int64, md5Sum, sha256Sum []byte, metadata map[string]string) (ObjectInfo, error) {
-	return c.putObjectDo(context.Background(), bucket, object, data, md5Sum, sha256Sum, size, PutObjectOptions{UserMetadata: metadata})
+	delKey := func(m map[string]string, key string) {
+		delete(m, strings.ToLower(key))
+		delete(m, strings.ToUpper(key))
```

Comment:
will miss key in mixed-case.

Reply:
can you give me an example?

Reply:
example - if  "Content-Type" is a key in the metadata map, the above code will not delete the entry.

Reply:
Oops yeah i should have used http.CanonicalHeaderKey()